### ISSUE TITLE
Improve algorithms for incremental/partial-fit training in streaming environment 

### ISSUE THREAD

**Issue by cfregly** [fluxcapacitor/pipeline#50]
currently, using Matrix Factorization with Gradient Descent Optimization.  better algos exist.

**Comment by cfregly** [comment]
http://quinonero.net/Publications/predicting-clicks-facebook.pdf

**Comment by cfregly** [comment]
Investigate fold-in method from @srowen:  https://www.slideshare.net/mobile/srowen/big-practical-recommendations-with-alternating-least-squares (slide 14).  

also, investigate the latest advancements with his oryx project:  http://oryx.io/

**Comment by cfregly** [comment]
Netflix's Offline, Nearline, and Online architecture:  http://techblog.netflix.com/2013/03/system-architectures-for.html

![netflix-nearline-recommendations](https://cloud.githubusercontent.com/assets/1438064/17529589/c428ce12-5e28-11e6-9f52-b813fd386676.png)

**Comment by cfregly** [comment]
this includes both Spark ML and TensorFlow, but we may split these into separate issues if needed

**Comment by cfregly** [comment]
Yahoo slides/talks:

http://www.slideshare.net/PatrickPletscher/training-largescale-ad-ranking-models-in-spark

**Comment by cfregly** [comment]
Great article by Uber on incremental data processing:  https://www.oreilly.com/ideas/ubers-case-for-incremental-processing-on-hadoop

**Comment by cfregly** [comment]
Interesting paper on incrementally training neural nets versus starting from scratch:  http://arxiv.org/abs/1511.05641

**Comment by cfregly** [comment]
Related to https://github.com/fluxcapacitor/pipeline/issues/114

**Comment by cfregly** [comment]
Spark Streaming + Spark ML Proposal:  
https://docs.google.com/document/d/1snh7x7b0dQIlTsJNHLr-IxIFgP43RfRV271YK2qGiFQ

Interesting reference from the above proposal:  http://huawei-noah.github.io/streamDM/

**Comment by cfregly** [comment]
part of larger effort
